src/mbgl/style/layer.cpp

```diff
@@ -91,6 +91,59 @@ void Layer::setMaxZoom(float maxZoom) {
     observer->onLayerChanged(*this);
 }
 
+Value Layer::serialize() const {
+    mapbox::base::ValueObject result;
+    result.emplace("id", getID());
```

Comment:
Could that affect binary size as invokes pair template constructor (https://en.cppreference.com/w/cpp/container/unordered_map/emplace) ?

Reply:
changed to make_pair